src/Swashbuckle.AspNetCore.SwaggerGen/XmlComments/XmlCommentsSchemaFilter.cs

```diff
@@ -39,21 +39,41 @@ private void ApplyMemberTags(OpenApiSchema schema, SchemaFilterContext context)
             var fieldOrPropertyMemberName = XmlCommentsNodeNameHelper.GetMemberNameForFieldOrProperty(context.MemberInfo);
             var fieldOrPropertyNode = _xmlNavigator.SelectSingleNode($"/doc/members/member[@name='{fieldOrPropertyMemberName}']");
 
-            if (fieldOrPropertyNode == null) return;
+            var recordTypeName = XmlCommentsNodeNameHelper.GetMemberNameForType(context.MemberInfo.DeclaringType);
+            var recordDefaultConstructorProperty =
+                _xmlNavigator.SelectSingleNode($"/doc/members/member[@name='{recordTypeName}']/param[@name='{context.MemberInfo.Name}']");
 
-            var summaryNode = fieldOrPropertyNode.SelectSingleNode("summary");
-            if (summaryNode != null)
-                schema.Description = XmlCommentsTextHelper.Humanize(summaryNode.InnerXml);
+            if (recordDefaultConstructorProperty != null)
+            {
+                var summaryNode = recordDefaultConstructorProperty.Value;
+                if (summaryNode != null)
+                    schema.Description = XmlCommentsTextHelper.Humanize(summaryNode);
+
+                var example = recordDefaultConstructorProperty.GetAttribute("example", string.Empty);
+                SetExample(schema, context, example);
+            }
 
-            var exampleNode = fieldOrPropertyNode.SelectSingleNode("example");
-            if (exampleNode != null)
+            if (fieldOrPropertyNode != null)
             {
-                var exampleAsJson = (schema.ResolveType(context.SchemaRepository) == "string") && !exampleNode.Value.Equals("null")
-                    ? $"\"{exampleNode.ToString()}\""
-                    : exampleNode.ToString();
+                var summaryNode = fieldOrPropertyNode.SelectSingleNode("summary");
+                if (summaryNode != null)
+                    schema.Description = XmlCommentsTextHelper.Humanize(summaryNode.InnerXml);
 
-                schema.Example = OpenApiAnyFactory.CreateFromJson(exampleAsJson);
+                var exampleNode = fieldOrPropertyNode.SelectSingleNode("example");
+                SetExample(schema, context, exampleNode?.Value);
             }
         }
+
+        private static void SetExample(OpenApiSchema schema, SchemaFilterContext context, string example)
```

Comment:
Maybe rename this to `TrySetExample()` as it might not actually do anything?

Reply:
Sure